docs/docs/integrations/tools/oxylabs.ipynb

```diff
@@ -0,0 +1,308 @@
+   "id": "69ea8139f6152b48",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_oxylabs import OxylabsSearchAPIWrapper, OxylabsSearchRun\n",
+    "\n",
+    "oxylabs_wrapper = OxylabsSearchAPIWrapper()\n",
+    "tool_ = OxylabsSearchRun(wrapper=oxylabs_wrapper)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "636efff6267b9bc1",
+   "metadata": {},
+   "source": "## Invocation"
+  },
+  {
+   "cell_type": "markdown",
+   "id": "272f5fdaed5fa000",
+   "metadata": {},
+   "source": "### Invoke directly with args"
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3c26724911f6cfdf",
+   "metadata": {},
+   "source": "The `OxylabsSearchRun` tool takes a single \"query\" argument, which should be a natural language query and returns combined string format result:"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b5f4d423cb3abf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tool_.invoke({\"query\": \"Restaurants in Paris.\"})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "784aad18bba31062",
+   "metadata": {},
+   "source": "### Invoke with ToolCall"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8afd1652e2c4e7c9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tool_ = OxylabsSearchRun(\n",
+    "    wrapper=oxylabs_wrapper,\n",
+    "    kwargs={\n",
+    "        \"result_categories\": [\n",
+    "            \"local_information\",\n",
+    "            \"combined_search_result\",\n",
+    "        ]\n",
+    "    },\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1b903fb2f8272df2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pprint import pprint\n",
+    "\n",
+    "model_generated_tool_call = {\n",
+    "    \"args\": {\n",
+    "        \"query\": \"Visit restaurants in Vilnius.\",\n",
+    "        \"geo_location\": \"Vilnius,Lithuania\",\n",
+    "    },\n",
+    "    \"id\": \"1\",\n",
+    "    \"name\": \"oxylabs_search\",\n",
+    "    \"type\": \"tool_call\",\n",
+    "}\n",
+    "tool_call_result = tool_.invoke(model_generated_tool_call)\n",
+    "\n",
+    "# The content is a JSON string of results\n",
+    "pprint(tool_call_result.content)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a436fc39ce17b6da",
+   "metadata": {},
+   "source": "## Chaining"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a6f23e65092e8d27",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Install the required libraries\n",
+    "%pip install --upgrade --quiet langchain langchain-openai"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "21078a0c265759ff",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import getpass\n",
+    "import os\n",
+    "\n",
+    "from langchain import hub\n",
+    "from langchain.agents import AgentExecutor, create_tool_calling_agent\n",
```

Comment:
Use the langgraph prebuilt agent rather than the one in langchain.

For example, 

https://python.langchain.com/docs/integrations/tools/tavily_search/#use-within-an-agent